packages/cli-e2e/utils/login.ts

```diff
@@ -0,0 +1,109 @@
+import {Page} from 'puppeteer-core';
+import type {Browser} from 'puppeteer-core';
+import {ChildProcessWithoutNullStreams, spawn} from 'child_process';
+import {answerPrompt, CLI_EXEC_PATH, isYesNoPrompt} from './cli';
+
+function isLoginPage(page: Page) {
+  return page.url().match(/https:\/\/platform.*cloud\.coveo\.com\/login/);
+}
+
+function isLoggedIn(page: Page) {
```

Comment:
Will the user alway be redirected to the "redirect uri" page when logged in?
There probably a better way to let the test know it is logged in.

Reply:
One thing we could do is login once at the beginning of the tests suite, and keep the token in the keychain.

So the `loggedIn` state would actually be "is the token available in the keychain".

But I understand what you mean with the final redirection.

I think it's mostly a misunderstanding with the function name.

`isLoginFlowFinished` might be better ?

Reply:
I think logging in at the beginning of the test suite is a good idea and also much cleaner. However, it causes the test in `auth.spec.ts` to fail since the redirection to the login page is not happening when already logged in (automatically redirected to the redirect uri).

I was thinking about these 2 options:
1. for every test file, login in the setup phase and logout in the teardown phase
2. discard the `auth.spec.ts` since the `auth:login` command is already  tested other files

Anyway, let me know what you think...